### PR TITLE
Add OVAL and BIG_OVAL tracks.

### PR DESCRIPTION
**Pull request by mikelehen**
* Separate tracks out into their own tracks.ts file. 
* Add OVAL and BIG_OVAL tracks.
* Make OVAL the default.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -18,9 +18,9 @@
     <canvas id="track"></canvas>
 
     <script type="module">
-        import { Racetrack, U_TRACK } from './src/racetrack.ts';
+        import { Racetrack, U_TRACK, BIG_OVAL, OVAL } from './src/racetrack.ts';
 
-        const rt = new Racetrack(document.getElementById(`track`), U_TRACK);
+        const rt = new Racetrack(document.getElementById(`track`), OVAL);
```
Comment (mikelehen) [inline index.html:23]: I can put it back to U_TRACK if you prefer. 🤷

File: src/racetrack.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Point, linePoints, add, sub, scale, round, ceil, id, pointFromId, neighbors } from './points.js';
-export { Racetrack, U_TRACK };
+import { Track, U_TRACK, OVAL, BIG_OVAL } from './tracks.js';
```
Comment (mikelehen) [inline src/racetrack.ts:2]: I factored the tracks out into their own file.

File: src/tracks.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    grid: 20,
+    startLine: [[400, 10], [400, 110]],
+    finishLine: [[380, 10], [380, 110]],
+    trackWidth: 100,
+    path: [[400, 60], [740, 60], [740, 340], [60, 340], [60, 60], [380, 60]],
+}
+
+export const BIG_OVAL:Track = {
+    dim: [800, 400],
+    grid: 10,
```
Comment (mikelehen) [inline src/tracks.ts:34]: This is similar to OVAL but has a grid size of 10 so it ends up being a longer course.

File: src/racetrack.ts [hunk 4]
```diff
@@ -171,15 +152,17 @@ class Racetrack {
...
-        this.ctx.fillStyle = 'black';
-        this.ctx.font = '12px sans-serif';
-        this.ctx.textAlign = 'center';
-        this.ctx.textBaseline = 'middle';
-        for (let point of this.gridPoints()) {
-            const [x, y] = scale(this.track.grid, point);
-            const pos = id(point);
-            if (this.finishDistances.has(pos)) {
-                this.ctx.fillText(this.finishDistances.get(pos)!.toString(), x, y);
+        if (this.track.grid >= 20) {
```
Comment (mikelehen) [inline src/racetrack.ts:155]: The numbers get messy if the grid size is too small.

File: src/racetrack.ts [hunk 5]
```diff
@@ -198,9 +181,10 @@ class Racetrack {
     }
 
     dot(x: number, y: number, color: string) {
+        const radius = this.track.grid / 4;
```
Comment (mikelehen) [inline src/racetrack.ts:184]: Make the dot size adaptive based on the grid spacing...

**Comment by mckoss** [comment]
New tracks - Passed tests - looks good.

**Pull request merged**